### PR TITLE
Add warm up steps for CosineDecay

### PR DESCRIPTION
**Pull request by fsx950223**
Sometimes,we may use freeze training or warm up training,so I add warm up steps to ensure we can get correct learning rate.And CosineDecay init don't need global step,I fixed CosineDecay doc by the way.

### CODE REVIEW

File: tensorflow/python/keras/optimizer_v2/learning_rate_schedule.py [hunk 1]
```diff
@@ -601,10 +605,10 @@ def __call__(self, step):
           self.initial_learning_rate, name="initial_learning_rate")
       dtype = initial_learning_rate.dtype
       decay_steps = math_ops.cast(self.decay_steps, dtype)
-
+      warmup_steps = math_ops.cast(self.warmup_steps, dtype)
       global_step_recomp = math_ops.cast(step, dtype)
       global_step_recomp = math_ops.minimum(global_step_recomp, decay_steps)
-      completed_fraction = global_step_recomp / decay_steps
+      completed_fraction = (global_step_recomp - warmup_steps) / decay_steps
```
Comment (tanzhenyu) [inline tensorflow/python/keras/optimizer_v2/learning_rate_schedule.py]: this can be negative values? shouldn't we do tf.maximum(0, (global_step_recomp - warmup_steps))?

File: tensorflow/python/keras/optimizer_v2/learning_rate_schedule.py [hunk 2]
```diff
@@ -537,6 +537,7 @@ def __init__(
       self,
       initial_learning_rate,
       decay_steps,
+      warmup_steps=0,
```
Comment (tanzhenyu) [inline tensorflow/python/keras/optimizer_v2/learning_rate_schedule.py]: do you have a good example of where this is used? API changes are non-trivial and needs to be closely examined.
